pkg/controller/plan/validation.go

```diff
@@ -792,14 +814,37 @@ func (r *Reconciler) validateVddkImage(plan *api.Plan) (err error) {
 			r.Log.Info("validation of VDDK job is in progress", "image", image)
 			plan.Status.SetCondition(vddkValidationInProgress)
 		}
+		pods := &core.PodList{}
+		listerr := r.List(context.TODO(), pods, &client.ListOptions{
+			Namespace:     plan.Spec.TargetNamespace,
+			LabelSelector: k8slabels.SelectorFromSet(map[string]string{"job-name": job.Name}),
+		})
+		if listerr == nil {
+			for _, pod := range pods.Items {
+				if strings.Contains(pod.Name, job.Name) {
+					// there should only be a single init container
+					initStatus := pod.Status.InitContainerStatuses[0]
+					if initStatus.State.Waiting != nil {
+						plan.Status.SetCondition(vddkImageNotReady)
+					} else {
+						plan.Status.DeleteCondition(VDDKImageNotReady)
+					}
+				}
+			}
+		}
 		for _, condition := range job.Status.Conditions {
 			switch condition.Type {
 			case batchv1.JobComplete:
 				r.Log.Info("validate VDDK job completed", "image", image)
 				err = nil
 				return
 			case batchv1.JobFailed:
-				plan.Status.SetCondition(vddkInvalid)
+				if plan.Status.FindCondition(VDDKImageNotReady) != nil {
+					plan.Status.DeleteCondition(VDDKImageNotReady)
+					plan.Status.SetCondition(vddkImageNotFound)
```

Comment:
do I get this correctly that in this scenario (in which the VDDK image is not found), the pod will be set with `initStatus.State.Waiting != nil`?

Reply:
This is much needed!! I am struggling to understand what caused this sudden invalidvddkimage error on my vcenter running 7.x and I am using vddk 8.0.2 from broadcom - I queried the controller inventory / main logs - it just tells me InvalidVDDKImage - Steps tried

1. Deleted and recreated vsphere provider with diff. images of vddk - from quay.io/kubevirt/vddk , from broadcom site - vddk 7.03.8.0.2,8.0.3/// 
2. Deleted plans and recreated them
3. Deleted controller pods and updated deployment to have more cpu & mem. resources... below error from provider with - vddk:8.0.2-22388865

<h2 class="co-section-heading " style="padding: 0px; margin: 0px 0px 20px; box-sizing: border-box; font-size: var(--pf-v5-global--FontSize--xl); font-weight: var(--pf-v5-global--FontWeight--normal); font-family: var(--pf-v5-global--FontFamily--heading); line-height: var(--pf-v5-global--LineHeight--md); word-break: break-word; display: flex; justify-content: space-between; color: rgb(255, 255, 255); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span style="box-sizing: border-box;">Conditions</span></h2>
Type | Status | Updated | Reason | Message


```
--
{"level":"info","ts":"2024-07-24 07:37:30.842","logger":"plan|vvqrj","msg":"Condition added.","plan":{"name":"testmigration1","namespace":"konveyor-forklift"},"condition":{"type":"VDDKInvalid","status":"True","reason":"NotSet","category":"Critical","message":"VDDK init image is invalid","lastTransitionTime":null}}
{"level":"info","ts":"2024-07-24 07:37:30.842","logger":"plan|vvqrj","msg":"Condition deleted.","plan":{"name":"testmigration1","namespace":"konveyor-forklift"},"condition":{"type":"ValidatingVDDK","status":"True","reason":"Started","category":"Advisory","message":"Validating VDDK init image","lastTransitionTime":"2024-07-24T07:32:30Z"}}
{"level":"debug","ts":"2024-07-24 07:37:30.842","logger":"events","msg":"VDDK init image is invalid","type":"Warning","object":{"kind":"Plan","namespace":"konveyor-forklift","name":"testmigration1","uid":"a92ed2c6-1e2f-4725-8b34-306a6ad3e87d","apiVersion":"forklift.konveyor.io/v1beta1","resourceVersion":"136041944"},"reason":"VDDKInvalid"}
{"level":"debug","ts":"2024-07-24 07:37:30.842","logger":"events","msg":"Validating VDDK init image","type":"Normal","object":{"kind":"Plan","namespace":"konveyor-forklift","name":"testmigration1","uid":"a92ed2c6-1e2f-4725-8b34-306a6ad3e87d","apiVersion":"forklift.konveyor.io/v1beta1","resourceVersion":"136041944"},"reason":"ValidatingVDDK"}
{"level":"info","ts":"2024-07-24 07:37:30.851","logger":"plan|vvqrj","msg":"Reconcile ended.","plan":{"name":"testmigration1","namespace":"konveyor-forklift"},"reQ":0}
```
-- | -- | -- | -- | --
TargetNameNotValid | True | Jul 24, 2024, 1:02 PM | NotValid | Target VM name does not comply with DNS1123 RFC, will be automatically changed.
VDDKInvalid | True | Jul 24, 2024, 1:07 PM | NotSet | VDDK init image is invalid

Reply:
How can I use this file or the updates you have made, i am running forklift operator 2.7 version

Reply:
@jonner I'm not absolutely sure about that but I think that the error may also be related to scheduling (when the pod cannot be scheduled to a node) but as we don't set resource requests, it practically shouldn't happen. so I'm good with your approach

Reply:
@jsakil14 Forklift 2.7 has not been released yet so if you're running 2.7 it means you use "latest" images and those will be updated when we merge this PR

Reply:
> @jonner I'm not absolutely sure about that but I think that the error may also be related to scheduling (when the pod cannot be scheduled to a node) but as we don't set resource requests, it practically shouldn't happen. so I'm good with your approach

Well, it's easy enough to check the reason for ErrImagePull to differentiate those failure cases if you think it would be useful.

Reply:
> @jsakil14 Forklift 2.7 has not been released yet so if you're running 2.7 it means you use "latest" images and those will be updated when we merge this PR

I meant operator version - v2.7.0 - Also , openshift console - installed operator gives below : Forklift Operator
2.7.0 provided by Konveyor @ahadas

Reply:
> > @jsakil14 Forklift 2.7 has not been released yet so if you're running 2.7 it means you use "latest" images and those will be updated when we merge this PR
> 
> I meant operator version - v2.7.0 - Also , openshift console - installed operator gives below : Forklift Operator 2.7.0 provided by Konveyor @ahadas

yeah, but you can see [here](https://github.com/redhat-openshift-ecosystem/community-operators-prod/tree/main/operators/forklift-operator) that we didn't publish 2.7.0 in the operator hub so if you managed to install it, I assume you've added another catalog source and the 'latest' tag of the images is used (you can check that by looking at the image of any pod in the konveyor-forklift namespace) - and then what I wrote about would hold

Reply:
> > @jonner I'm not absolutely sure about that but I think that the error may also be related to scheduling (when the pod cannot be scheduled to a node) but as we don't set resource requests, it practically shouldn't happen. so I'm good with your approach
> 
> Well, it's easy enough to check the reason for ErrImagePull to differentiate those failure cases if you think it would be useful.

ok, so let's do that then

Reply:
> > > @jsakil14 Forklift 2.7 has not been released yet so if you're running 2.7 it means you use "latest" images and those will be updated when we merge this PR
> > 
> > 
> > I meant operator version - v2.7.0 - Also , openshift console - installed operator gives below : Forklift Operator 2.7.0 provided by Konveyor @ahadas
> 
> yeah, but you can see [here](https://github.com/redhat-openshift-ecosystem/community-operators-prod/tree/main/operators/forklift-operator) that we didn't publish 2.7.0 in the operator hub so if you managed to install it, I assume you've added another catalog source that uses the 'latest' tag (you can check that by looking at the image of any pod in the konveyor-forklift namespace) - and then what I wrote about would hold

Looks like I do use **latest**...Do I have to change it? @ahadas 

```
 env:
    - name: WATCH_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.annotations['olm.targetNamespaces']
    - name: ANSIBLE_GATHERING
      value: explicit
    - name: CONTROLLER_IMAGE
      value: quay.io/kubev2v/forklift-controller:latest
    - name: API_IMAGE
      value: quay.io/kubev2v/forklift-api:latest
    - name: MUST_GATHER_IMAGE
      value: quay.io/kubev2v/forklift-must-gather:latest
    - name: UI_PLUGIN_IMAGE
      value: quay.io/kubev2v/forklift-console-plugin:latest
    - name: VALIDATION_IMAGE
      value: quay.io/kubev2v/forklift-validation:latest
    - name: VIRT_V2V_IMAGE
      value: quay.io/kubev2v/forklift-virt-v2v:latest
    - name: VIRT_V2V_WARM_IMAGE
      value: quay.io/kubev2v/forklift-virt-v2v-warm:latest
    - name: VIRT_V2V_DONT_REQUEST_KVM
    - name: SNAPSHOT_REMOVAL_TIMEOUT
    - name: SNAPSHOT_STATUS_CHECK_RATE
    - name: CLEANUP_RETRIES
    - name: CDI_EXPORT_TOKEN_TTL
    - name: FILESYSTEM_OVERHEAD
    - name: BLOCK_OVERHEAD
    - name: POPULATOR_CONTROLLER_IMAGE
      value: quay.io/kubev2v/populator-controller:latest
    - name: OVIRT_POPULATOR_IMAGE
      value: quay.io/kubev2v/ovirt-populator:latest
    - name: OPENSTACK_POPULATOR_IMAGE
      value: quay.io/kubev2v/openstack-populator:latest
    - name: OVA_PROVIDER_SERVER_IMAGE
      value: quay.io/kubev2v/forklift-ova-provider-server:latest
    - name: OVIRT_OS_MAP
    - name: VSPHERE_OS_MAP
    - name: OPERATOR_CONDITION_NAME
      **value: forklift-operator.v2.7.0**
    **image: quay.io/kubev2v/forklift-operator:latest**
    imagePullPolicy: Always
    livenessProbe:
      failureThreshold: 3
      httpGet:
        path: /healthz
        port: 6789
        scheme: HTTP
      initialDelaySeconds: 15
      periodSeconds: 20
      successThreshold: 1
      timeoutSeconds: 1
    name: forklift-operator
```

Reply:
#971